eZ/Publish/Core/FieldType/Image/ImageStorage/Gateway.php

```diff
@@ -59,6 +59,11 @@ abstract public function removeImageReferences($uri, $versionNo, $fieldId);
      */
     abstract public function countImageReferences($uri);
 
+    /**
+     * Returns if there is reference to the given $uri.
+     */
+    abstract public function isImageReferenced(string $uri): bool;
```

Comment:
Thought about this, not sure if this is part of BC policy, but if we can be safe about it then I do not see why we shouldnt.

Reply:
I wouldn't fallback to the `countImageReferences($uri)` as we already know it is very flawed that may (and most likely will) result with the very same issue we're trying to resolve right now: 200s of SQL queries time. 

```php
public function isImageReferenced(string $uri): bool
{
    throw new NotImplementedException(__METHOD__);
}
```

Would be much safer. WDYT @Steveb-p ?

Reply:
This would postpone the issue of missing code until runtime, so :-1: 

We should either provide this as a preventative fallback against a BC break, or not at all.

My proposal of including a default implementation that does exactly what happens currently is only to prevent extending implementation from failing (as it will force dev-users to adjust the extending class, which constitutes a BC break).

I've additionally suggested adding a `trigger_error` with `E_USER_DEPRECATION` notice (with removal in 4.x I guess), so it won't go under the radar.

Reply:
> This would postpone the issue of missing code until runtime, so 👎
>
> We should either provide this as a preventative fallback against a BC break, or not at all.

@Steveb-p But this is new method, so the only BC we must provide **IS** PHP class inheritance on an abstract class. But if we do it in a way of providing current implementation by default, we're introducing the very same issue we're trying to resolve.

Reply:
After private discussion with @alongosz, as there is no BC policy on that we decided to drop default implementation and fix the old method instead of creating new one.
@Nattfarinn